backend/hardware/gpiochip/core.py

```diff
@@ -3,16 +3,25 @@
 and the individual files for configuration information.
 """
 
+from hardware.gpiochip.gpiod_simulation import GPIODChipSimulation
+from hardware.gpiochip.gpiod_chipset import GPIODChipset
+
+IMPORT_ERROR = False
+
+# We check for an import error here as this ikmport requires the gpiod python library
+# to be installed, which is not a requirement
+try:
+    from hardware.gpiochip.gpiod import GPIODChip
+except ModuleNotFoundError:
+    IMPORT_ERROR = True
+
 
 def createGPIOChip(gpioConfig: dict, devices: dict):
     gpioType = gpioConfig['implementation']
-    if gpioType == "gpiod":
-        from hardware.gpiochip.gpiod import GPIODChip
+    if gpioType == "gpiod" and not IMPORT_ERROR:
         return GPIODChip(gpioConfig)
     if gpioType == "simulation":
-        from hardware.gpiochip.gpiod_simulation import GPIODChipSimulation
         return GPIODChipSimulation(gpioConfig)
     if gpioType == "gpiod_chipset":
-        from hardware.gpiochip.gpiod_chipset import GPIODChipset
         return GPIODChipset(gpioConfig, devices)
     raise Exception("Unsupported gpiochiptype")
```

Comment:
(Copied from #267)

This comment goes for all of the backend/hardware/x/core.py import changes:

I think it would be best to leave the imports inside of the function calls in this case. Any hardware used may have additional python libraries and/or system configuration dependencies, and only trying to import them when someone is actually trying to use them rather than catching and keeping track of every possible error that could occur on every possible import is a lot of overhead. This block of code for instance will not display to the user as to what the real source of the error is if they are attempting to use a configuration that has "gpiod" but without the system library installed for it. It will instead throw the exception "Unsupported gpiochiptype", which is inaccurate because that is a supported value for gpiochiptype, instead of the more meaningful ModuleNotFoundError.

Reply:
Fair enough, I think this is one of those edge cases where inline imports can be ok but I wouldn't want to do any more than the import on type attempt